Add MVS user instructions (#229)
rl-institut/mvs_eland: docs/simulating_with_the_mvs.rst

The path to the input folder is up to you.

.. image:: images/folder_structure_inputs.png
 :width: 200

There are two options to insert all input data – Json and CSV. These will be explained below.

Json file: mvs_config.json
##########################

In combination especially with the planned graphical user interface for the MVS (EPA),
a json file will be used to store and provide all input data necessary to understand relation.
The Json file itself is created by the EPA, ie. there are no manual changes.
You can use a specific Json input file if you want to test a simulation that has been made public online,
one test simulation, or as a developer that has knowingly edited the Json file.

As this requires to adhere to quite specific formatting rules,
this can really only be recommended for advanced users.

There can only be a single Json file in your input folder.
As some parameters in the Json file link to a time series provided as a CSV,
the folder "time_series" should be present in your input folder
and provide all necessary input data that is an timeseries format.
This can include for example PV generation time series demand time series.

> Csv files: csv_elements folder

Reviewer: I would start with explaining csv and then json, as users, that are not developers can find their way faster.

Author: True, will change that next.

##############################

> Usually user, that is not using the MVS in combination with the EPA,

Reviewer: ```suggestion
Usually a user, that is not using the MVS in combination with the EPA,
```

- `energyStorage.csv <https://github.com/rl-institut/mvs_eland/blob/dev/inputs/csv_elements/energyStorage.csv>`_ - List of energy storages of the energy system
- `storage_01.csv <https://github.com/rl-institut/mvs_eland/blob/dev/inputs/csv_elements/storage_01.csv>`_ - Technical parameters of each energy system
- `fixcost.csv <https://github.com/rl-institut/mvs_eland/blob/dev/inputs/csv_elements/fixcost.csv>`_ - fix project development/maintenance costs (should not be used currently)
- `simulation_settings.csv <https://github.com/rl-institut/mvs_eland/blob/dev/inputs/csv_elements/simulation_settings.csv>`_ - Simulation settings, including start date and duration
- `project_data.csv <https://github.com/rl-institut/mvs_eland/blob/dev/inputs/csv_elements/project_data.csv>`_ - some generic project information

When defining your energy system with this CSV files,
please also refer to the definition of parameters that you can find here: `stable <https://mvs-eland.readthedocs.io/en/stable/MVS_parameters.html>`_ / `latest <https://mvs-eland.readthedocs.io/en/latest/MVS_parameters.html>`_.

Defining an energy system
-------------------------

> When you energy system you basically have to fill out the CSV sheets that are provided in the folder "csv_elements".

Reviewer: ```suggestion
For defining your energy system you basically have to fill out the CSV sheets that are provided in the folder "csv_elements".
```

> For each asset you want to add, you have to add new column.

Reviewer: ```suggestion
For each asset you want to add, you have to add a new column.
```

If you do not have an asset of a specific type,
simply leave the columns empty (but leave the columns with the parameter names and units).

The unit columns also tell you what type of information is required from you (string, boolean, number).

> In case doubt, also consider the parameter list that is linked above.

Reviewer: ```suggestion
In case of doubts, also consider the parameter list that is linked above.
```

> Do not delete any of the rows of the CSV´s – each parameters needed for the simulation.

Reviewer: ```suggestion
Do not delete any of the rows of the CSV´s – each parameter is needed for the simulation.
```

There will also be warnings if you do so.

Building a model from assets and energy flows
#############################################

Simulating an energy system with the MVS requires a certain level of abstraction.
In general, as it is based on the programming framework oemof,
it will depict the energy system only as linearized model.
This allows for the quick computation of the optimal system sizing and approximate dispatch,
but does not replace operational management.

The level of abstraction and system detail needed for an MVS simulation will be explained based on an exemplary local energy system.
Let's assume that we want to simulate an industrial site with some electrical demand, the grid connection, a battery as well as a PV plant.

> A schematic of such a system shown below.

Reviewer: ```suggestion
A schematic of such a system is shown below.
```

As such, in reality, the battery may be on an own DC electricity bus,
which is either the separate from or identical to the DC bus of the PV plant.
Both DC busses would have to be interconnected with the main electricity bus (AC) through an inverter,
or in case of bi-directional flow for the battery with an rectifier as well.

Just like so, the DSO could either be only providing electricity also allowing feed in,
or the demand may be split up into multiple demand profiles.
This granularity of information would be something that the MVS model requires to properly depict the system behaviour and resulted optimal capacities and dispatch.
The information fed into the MVS via the CSV's would therefore define following components:

.. image:: images/energy_system_model.png
 :width: 200

Ideally you scratch down the energy system you want to simulate with the above-mentioned granularity
and only using sources, sinks, transformers and buses (meaning the oemof components).
When interconnecting different assets make sure that you use the correct bus name in each of the CSV input files.
The bus names are defined with *input_direction* and *output_direction*.

> If you interconnect your assets or buses incorrectly the system will still be bought but the simulation terminated.

Reviewer: Not sure if you wanted to say that the simulation will not be terminated in this case.
What do you mean by "bought" by the way?

Author: ```suggestion
If you interconnect your assets or buses incorrectly the system will still be built but the simulation terminated.
```

If you're not sure whether or not you build your system correctly change the parameter *plot_networkx_graph* in the simulation_settings to True.

> When executing the simulation, the MVS were now generate a rough graphic visualisation of your energy system.

Reviewer: ```suggestion
When executing the simulation, the MVS will now generate a rough graphic visualisation of your energy system.
```

> There, all components and buses should be part of a single system - otherwise you misconfigured your energy system.

Reviewer: ```suggestion
There, all components and buses should be part of a single system (i.e. linked to each other) - otherwise you misconfigured your energy system.
```
